### PR TITLE
Kriging fix

### LINKED ISSUE #2403: KrigingAlgorithm fails if basis is empty
### What happened?

If we provide an empty basis to the kriging algorithm, it fails!

We also need to have a ctor without he `basis` argument

### How to reproduce the issue?

```shell
ot.RandomGenerator.SetSeed(0)
model = ot.AbsoluteExponential()
size = 15
x = ot.Normal().getSample(size)
y = x + ot.Normal(0, 0.01).getSample(size)
algo = ot.KrigingAlgorithm(x, y, model, ot.Basis())
algo.run()
```

### Version

1.21

### Operating System

all

### Additional Context

The bug is recent (`1.21`) and not appearing in oldest versions

### PR DESCRIPTION
**Pull request by sofianehaddad**
Fix #2403

### CODE REVIEW

File: python/src/KrigingAlgorithm_doc.i.in [hunk 1]
```diff
@@ -5,8 +5,7 @@ Refer to :ref:`kriging`.
 
 Available constructors:
     KrigingAlgorithm(*inputSample, outputSample, covarianceModel, basis*)
-
-    KrigingAlgorithm(*inputSample, outputSample, covarianceModel, basisCollection*)
+    KrigingAlgorithm(*inputSample, outputSample, covarianceModel*)
```
Comment (jschueller) [inline python/src/KrigingAlgorithm_doc.i.in:8]: no need for a ctor list when they are nested, I think you can remove the whole list and just mention that the basis argument is optional

Comment (sofianehaddad) [reply python/src/KrigingAlgorithm_doc.i.in:8]: ok

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/openturns/KrigingAlgorithm.hxx [hunk 2]
```diff
@@ -53,6 +53,11 @@ public:
                     const CovarianceModel & covarianceModel,
                     const Basis & basis);
 
+  /** Constructor */
+  KrigingAlgorithm(const Sample &inputSample,
+                   const Sample &outputSample,
+                   const CovarianceModel &covarianceModel);
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/Kriging/openturns/KrigingAlgorithm.hxx:59]: I think its enough to make basis optional in the previous ctor:
```const Basis & basis = Basis()```
then it looks like GLM will handle this

Comment (sofianehaddad) [reply lib/src/Uncertainty/Algorithm/MetaModel/Kriging/openturns/KrigingAlgorithm.hxx:59]: ok

File: python/src/KrigingAlgorithm_doc.i.in [hunk 3]
```diff
@@ -6,8 +6,6 @@ Refer to :ref:`kriging`.
 Available constructors:
     KrigingAlgorithm(*inputSample, outputSample, covarianceModel, basis*)
```
Comment (jschueller) [inline python/src/KrigingAlgorithm_doc.i.in:7]: this can be dropped

**Review by jschueller — APPROVED** [review]

**Review by regislebrun — APPROVED** [review]

**Pull request merged**